instrumentation/graphql/lib/opentelemetry/instrumentation/graphql/patches/opentelemetry_graphql_tracing.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module GraphQL
+      module Patches
+        # OpenTelemery Graphql platform_tracing implementation for instrumentation
+        class OpenTelemetryGraphQLTracing < ::GraphQL::Tracing::PlatformTracing
+          self.platform_keys = {
+            'lex' => 'lex.graphql',
+            'parse' => 'parse.graphql',
+            'validate' => 'validate.graphql',
+            'analyze_query' => 'analyze.graphql',
+            'analyze_multiplex' => 'analyze.graphql',
+            'execute_multiplex' => 'execute.graphql',
+            'execute_query' => 'execute.graphql',
+            'execute_query_lazy' => 'execute.graphql'
+          }
+
+          def platform_trace(platform_key, key, data)
```

Comment:
We additionally trace `execute_field` and `execute_field_lazy`, but rather than creating a span for each field, we append the fields to a `field_contexts` array hanging off the current `Span`, and monkey-patch `finish` to add the array as an attribute on the span. It's ugly, and I wonder if we can do it another way here. In general, it's useful to users to know which fields are evaluated as part of the query.

Reply:
ah, that's interesting. It sounds like valuable info imo, so i think we should try to support it. Let me see if i can find a more canonical way to do so.

Reply:
Shopify's tracing gem doesn't support Events, so I forgot about those. I think that might be a better way to handle `execute_field` and `execute_field_lazy`.